### ISSUE TITLE
MA0206 (unnecessary braces) fights against documentation

### ISSUE THREAD

**Issue by Timovzl** [meziantou/Meziantou.Analyzer#1196]
Hello again,

Analyzer [MA0206](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0206.md), "remove unnecessary braces in type declaration", can sometimes fight against documentation.

```cs
/// <summary>
/// I show up when you hover my constructor invocation too!
/// </summary>
public sealed class Documented()
{
}

/// <summary>
/// I don't. :(
/// </summary>
public sealed class HalfDocumented
{
}
```

You might think of taking into consideration the presence of documentation on the type and the absence of it on its constructors, but... then consistency comes into play.

If I have some types with summaries, all using an empty primary constructor for this very reason, and they are mixed with similar types that just so happen _not_ to need any documentation, then I will still treat them identically, i.e. all should have a primary constructor. An unsuspecting developer might need to add a summary later! The same applies when some types take parameters and others do not. Consistency is valuable.

As it stands, the analyzer fights that consistency by insisting that an empty primary constructor is objectively worse. It's not. There is too much to consider for the analyzer to figure out what's best contextually. It should not steer us _away_ from good practice.

In fact, if the parentheses are there, they are likely there for a reason. A developer does not type them by accident. It takes more work to type them. Their presence indicates intent.

As always, I'm fully aware that I could disable the analyzer, but I respect the hell out of this package, so I fight to help make its adoption a no-brainer for as many folks as possible. :smile: I'd love to keep us miles away from "the SonarQube experience" (where too many false alarms cause you to doubt if the product is worth the hassle).

I hope my illustration has shown MA0206 to be opinionated enough to be made _opt-in_, meaningful only to those who both have a reason to dislike empty primary constructors and discourage them even when it sacrifices consistency.

**Comment by meziantou** [comment]
Thanks for reporting the issue. It should report everywhere braces are not needed.
